### PR TITLE
Fix on recent neovim build

### PR DESCRIPTION
**Pull request by jhchabran**
I was configuring my go env and struggled to get `dap-go.nvim` to work. I managed to fix the two issues that were blocking me and included them in this PR. 

I'm not sure of what is happening, but I thought that bringing them forward would be useful in case someone else is blocked or if you want to investigate. 

```
:version                                                                                                                                                                                                                                                                                                                                                                                                   
NVIM v0.7.0-dev+923-gecec95712                                                                                                                                                                                                                                                                                                                                                                             
Build type: Release                                                                                                                                                                                                                                                                                                                                                                                        
LuaJIT 2.1.0-beta3               
```

### CODE REVIEW

File: lua/dap-go.lua [hunk 1]
```diff
@@ -59,7 +59,7 @@ local function setup_adapter()
     local port = configuration.port or '38697'
     local addr = string.format('%s:%s', host, port)
 
-    if configuration.options.env then
+    if configuration.options and configuration.options.env then
```
Comment (jhchabran) [inline lua/dap-go.lua:62]: `configurations.options` was not present in my case and caused the common field not found when trying to access `env`.

File: lua/dap-go.lua [hunk 2]
```diff
@@ -76,6 +76,8 @@ local function setup_adapter()
       :start()
 
     vim.defer_fn(function()
+      -- I have no idea why this is required, I suspect the firewall or something else?
+      os.execute("sleep " .. tonumber(1))
```
Comment (jhchabran) [inline lua/dap-go.lua]: This is the definition of empirical. I suspect there is a timing issue here. I don't know what would be the best way to fix that.

Comment (yriveiro) [reply lua/dap-go.lua]: @jhchabran do you have the error message of this one?

You are adding a sleep instruction inside the defer of the callback. That should not be necessary as the defer is scheduled to run after the 100 ms timeout. 

I suspect that your system is taking longer to start the server. Could you use a value higher and see if the behaviour is fixed. If it works I believe that a better approach would be make the timeout configurable.

File: lua/dap-go.lua (hunk 1 above)
Comment (yriveiro) [reply lua/dap-go.lua:62]: lgtm

File: lua/dap-go.lua (hunk 2 above)
Comment (jhchabran) [reply lua/dap-go.lua]: > You are adding a sleep instruction inside the defer of the callback. That should not be necessary as the defer is scheduled to run after the 100 ms timeout.

Oh, I totally missed the later argument with the `100` ms timeout, that's much much better. Sloppy Sunday hacking 😆 

I just tried it again, but strangely I'm not able to reproduce it, even when lowering the timeout to 10ms. 

The error I was getting was the `ERRCONNREFUSED` from failing to reach the `dlv` server.

**Comment by jhchabran** [comment]
@yriveiro I'll clear this PR later today by removing the sloppy sleep and making the timeout configurable 👍

**Review by yriveiro — APPROVED** [review]

**Comment by yriveiro** [comment]
@jhchabran thanks for the contribution 😃

**Pull request merged**